core/auth/src/main/java/software/amazon/awssdk/auth/credentials/AwsCredentialsProviderChain.java

```diff
@@ -156,19 +170,43 @@ public interface Builder extends CopyableBuilder<Builder, AwsCredentialsProvider
         /**
          * Configure the credentials providers that should be checked for credentials, in the order they should be checked.
          */
-        Builder credentialsProviders(AwsCredentialsProvider... credentialsProviders);
+        Builder credentialsIdentityProviders(
+            Collection<? extends IdentityProvider<? extends AwsCredentialsIdentity>> credentialsProviders);
+
+        /**
+         * Configure the credentials providers that should be checked for credentials, in the order they should be checked.
+         */
+        default Builder credentialsProviders(AwsCredentialsProvider... credentialsProviders) {
+            return credentialsProviders((IdentityProvider<? extends AwsCredentialsIdentity>[]) credentialsProviders);
+        }
+
+        /**
+         * Configure the credentials providers that should be checked for credentials, in the order they should be checked.
+         */
+        default Builder credentialsProviders(IdentityProvider<? extends AwsCredentialsIdentity>... credentialsProviders) {
+            throw new UnsupportedOperationException();
+        }
+
+        /**
+         * Add a credential provider to the chain, after the credential providers that have already been configured.
+         */
+        default Builder addCredentialsProvider(AwsCredentialsProvider credentialsProvider) {
+            return addCredentialsProvider((IdentityProvider<? extends AwsCredentialsIdentity>) credentialsProvider);
+        }
 
         /**
          * Add a credential provider to the chain, after the credential providers that have already been configured.
          */
-        Builder addCredentialsProvider(AwsCredentialsProvider credentialsProviders);
+        default Builder addCredentialsProvider(IdentityProvider<? extends AwsCredentialsIdentity> credentialsProvider) {
+            throw new UnsupportedOperationException();
+        }
 
         AwsCredentialsProviderChain build();
     }
 
     private static final class BuilderImpl implements Builder {
         private Boolean reuseLastProviderEnabled = true;
-        private List<AwsCredentialsProvider> credentialsProviders = new ArrayList<>();
+        private List<IdentityProvider<? extends AwsCredentialsIdentity>> credentialsProviders = new ArrayList<>();
```

Comment:
Should we rename credentialsProviders variable to identityProviders ?

Reply:
These are still aws credentials providers. Just identityProviders could also confused if other identity types are allowed. To remove ambiguity we could use `awsCredentialsIdentityProviders` but I think the names doesn't matter much for private members. If both types were present in the same context, one might want to distinguish, but not needed here. Shorter is more readable. Also, the public methods use `credentialsProvider` still, so using the same name for member is fine.